### PR TITLE
chore: add esm examples for TS SDK

### PR DESCRIPTION
**Pull request by jjleng**
### Description
Many developers want to use TS SDK as an ESM. This example demonstrates how. ESM means ECMAScript modules. ECMAScript is the spec for Javascript. It added many useful features in recent years including the ECMAScript modules (ESM). In the past, NodeJs adopted CommonJS (CJS) modules. The syntax is `require("xxxx")`. There are some downsides with CJS including synchronous loading, not being tree-shaking friendly, etc. Good news is NodeJS start supporting ESM natively now. The syntax is `import xx from yyy`. Some browsers even support ESM natively. Also, React apps also use the ESM syntax. More developers start using ESM because of code compatibility, async loading, tree-shaking friendly, etc.

Aptos TS SDK is a dual-module SDK. It can be used in CJS apps and ESM apps. We also support IIFE bundles that can be downloaded from CDN. Developers do not have to use package management tools (npm, yarn) to use the SDK.

This PR demonstrates how to use the SDK in an ESM app.

### Test Plan
yarn test

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4321)

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
could we give more details about why esm? what is esm?

**Review by davidiw — COMMENTED** [review]
I might not be the right person to review this though

**Comment by gregnazario** [comment]
Let's get some of the more typescript native people to look at this, I don't know why esm is @kent-white

**Comment by jjleng** [comment]
Updated the PR description with some basic information about ESM.

**Review by zacharydenton — APPROVED** [review]

File: ecosystem/typescript/sdk/examples/typescript-esm/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "aptos": "1.3.13"
+  },
+  "devDependencies": {
+    "@types/node": "18.6.2",
+    "npm-run-all": "4.1.5",
```
Comment (banool) [inline ecosystem/typescript/sdk/examples/typescript-esm/package.json:19]: Doesn't look like we need this right now?

**Review by banool — APPROVED** [review]

**Pull request merged**